Library/Contributions/brew_bash_completion.sh

```diff
@@ -48,6 +48,8 @@ __brewcomp ()
 
     for s in $1; do
         __brewcomp_words_include "$s" && continue
+        # do not auto-complete "instal" abbreviation for "install"
+        [[ $s == instal ]] && continue
         list="$list$s$sep"
```

Comment:
Sorry, but this is totally the wrong place to do this. `__brewcomp` is used to generate completions for all kinds of arguments, not just the command argument. E.g., if there was a formula named `instal`, it would never be completed.

Instead, the filtering should be done on the list obtained from `brew commands`. May I suggest:
```patch
--- i/Library/Contributions/brew_bash_completion.sh
+++ w/Library/Contributions/brew_bash_completion.sh
@@ -579,7 +579,7 @@ _brew_uses ()

 _brew ()
 {
-    local i=1 cmd
+    local i=1 cmd cmds

     # find the subcommand
     while [[ $i -lt $COMP_CWORD ]]; do
@@ -600,7 +600,8 @@ _brew ()
     done

     if [[ $i -eq $COMP_CWORD ]]; then
-        __brewcomp "$(brew commands --quiet --include-aliases)"
+        cmds="$(brew commands --quiet --include-aliases)"
+        __brewcomp "${cmds/$'\n'instal$'\n'/$'\n'}"
         return
     fi
```

(This assumes that `instal` will never be the first item in the list.)